src/aws-elasticache/user-group.ts

```diff
@@ -0,0 +1,162 @@
+import { Construct } from 'constructs';
+import { IUser } from './user';
+import { Engine } from './util';
+
+/**
+ * An User Group
+ */
+export interface IUserGroup extends IResource {
+  /**
+   * The ARN of the user group.
+   *
+   * @attribute
+   */
+  readonly userGroupArn: string;
+  /**
+   * The ID of the user group.
+   *
+   * @attribute
+   */
+  readonly userGroupId: string;
+}
+
+/**
+ * Properties for defining an User Group.
+ */
+export interface UserGroupProps {
+  /**
+   * The ID of the user group.
+   *
+   * \`userGroupId\` can have up to 40 characters.
+   *
+   * \`userGroupId\` must consist only of alphanumeric characters or hyphens, with the first character as a letter, and it can't end with a hyphen or contain two consecutive hyphens.
+   *
+   * @default - auto generate
+   */
+  readonly userGroupId?: string;
+
+  /**
+   * The list of Iser that belong to the user group.
+   *
+   * `default` user is automatically added to the user group.
+   *
+   * @default - only `default` user added
+   */
+  readonly users?: IUser[];
+}
+
+/**
+ * Attributes for importing an User Group.
+ */
+export interface UserGroupAttributes {
+  /**
+   * The ID of the user group.
+   */
+  readonly userGroupId: string;
+}
+
+/**
+ * Represents an user group construct in AWS CDK.
+ *
+ * @example
+ * declare const user: User;
+ *
+ * const userGroup = new UserGroup(
+ *   stack,
+ *   'UserGroup',
+ *   {
+ *      user: [user],
+ *   },
+ * );
+ */
+export class UserGroup extends Resource implements IUserGroup {
+  /**
+   * Imports an existing user group from attributes
+   */
+  public static fromUserGroupId(scope: Construct, id: string, userGroupId: string): IUserGroup {
+    class Import extends Resource implements IUserGroup {
+      public readonly userGroupId = userGroupId;
+      public readonly userGroupArn = Stack.of(this).formatArn({
+        resource: 'elasticache',
+        service: 'usergroup',
+        resourceName: userGroupId,
+      });
+    }
+    return new Import(scope, id);
+  }
+
+  /**
+   * The ARN of the user group.
+   */
+  readonly userGroupArn: string;
+
+  /**
+   * The ID of the user group.
+   */
+  readonly userGroupId: string;
+
+  private readonly props: UserGroupProps;
+
+  private readonly users: IUser[];
+
+  constructor(scope: Construct, id: string, props: UserGroupProps) {
+    super(scope, id, {
+      physicalName:
+        props.userGroupId ??
+        Lazy.string({
+          produce: () => Names.uniqueResourceName(this, { maxLength: 40, allowedSpecialCharacters: '-' }).toLowerCase(),
+        }),
+    });
+    this.props = props;
+    this.users = this.props.users ?? [];
+
+    this.validateUserGroupId();
+
+    const userGroup = this.createResource(this, 'Resource', {
+      engine: Engine.REDIS,
+      userGroupId: this.physicalName,
+      userIds: Lazy.list({ produce: () => ['default', ...this.users.map(user => user.userId)] }),
```

Comment:
I made it so that the `default` is automatically added because it must be included.

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-elasticache-usergroup.html#cfn-elasticache-usergroup-userids

Reply:
Just for confirmation, is this correct? I thought the doc means that you have to include a user id whose username is `default`, not that user id itself is `default`.

Reply:
I apologize for my misunderstanding, but we need to include users with the user **name** `"default"` (not user id).
When deploying at L1 without username "default", I got the following error.

> ElastiCacheServerlessCacheStack | 21:09:59 | CREATE_FAILED        | AWS::ElastiCache::UserGroup           | UserGroup (UserGroupC755C9ED) Resource handler returned message: "User group needs to contain a user with the user name default. 

This is also described in the following docs:
https://docs.aws.amazon.com/AmazonElastiCache/latest/dg/Clusters.RBAC.html#Users-management

Therefore, it is not appropriate to automatically add users with user id "default". 
Instead, I think we need to implement validation to check whether a user with username "default" exists among the added users.

I'll fix it.

Reply:
I've redesigned property referencing [Redshift's](https://github.com/open-constructs/aws-cdk-library/blob/main/src/aws-redshiftserverless/namespace.ts#L68) `defaultIamRole` and `iamRoles` property.

* Add `defaultUser' property and validation to check `defaultUser` name is `default`.
* Change `users` property to required and add validation to check `users` include `defaultUser`.

Reply:
Thanks. 
I used the link as a reference to add validation and removed the defaultUser property.